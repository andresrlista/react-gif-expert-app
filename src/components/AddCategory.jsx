import { useCallback, useState } from 'react';
import PropTypes from 'prop-types';

/**
 * It's a form that takes a value from an input and passes it to a callback function
 * @returns A form with an input
 */
export const AddCategory = ({ onNewCategory }) => {
  const [categoryValue, setCategoryValue] = useState('');

  const onInputChangeCategory = useCallback(({ target }) => {
    setCategoryValue(target.value);
  }, []);

  const onSubmit = (e) => {
    e.preventDefault();
    if (categoryValue.trim().length <= 1) return;
    onNewCategory(categoryValue.trim());
    setCategoryValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        name='categoria'
        id='categoria'
        data-testid='categoria'
        type='text'
        value={categoryValue}
        placeholder='Search gifs...'
        onChange={onInputChangeCategory}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
