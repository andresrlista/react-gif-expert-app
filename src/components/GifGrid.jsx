import PropTypes from 'prop-types';
import { GifCard } from './index';
import useFetchGifs from '../hooks/useFetchGifs';

/* A functional component that is exporting a function called GifGrid. */
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);
  return (
    <>
      <h3>
        {category}
        <hr />
      </h3>
      {isLoading && <h2>Loading...</h2>}
      <div className='card-grid'>
        {images.map((image) => (
          <GifCard key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
