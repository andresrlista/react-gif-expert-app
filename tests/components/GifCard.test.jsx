import { getByRole, render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

describe('Test on GifCard', () => {
  const title = 'test';
  const url = 'https://url.test.com/';

  test('Must match snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('title must match with url', () => {
    render(<GifCard title={title} url={url} />);
    const { src, alt } = screen.getByRole('img');
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test('Must show the title', () => {
    render(<GifCard title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
