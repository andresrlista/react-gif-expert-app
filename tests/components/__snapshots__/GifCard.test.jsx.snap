// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifCard Must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://url.test.com/"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
